src/L2DebtManager.sol

```diff
@@ -0,0 +1,183 @@
+    IERC20 public USDC;
+
+    uint256 public liquidationThreshold;
+
+    address public etherFiCashSafe;
+
+    uint256 private eEthPriceInUSDC; // todo: replace it with Oracle
+
+
+    event SuppliedUSDC(uint256 amount);
+    event DepositedCollateral(address indexed user, uint256 amount);
+    event Borrowed(address indexed user, uint256 borrowUsdcAmount);
+    event RepaidWithUSDC(address indexed user, uint256 repaidUsdcDebtAmount);
+    event RepaidWithEETH(address indexed user, uint256 repaidUsdcDebtAmount, uint256 repaidCollateralEEthAmount);
+    event Liquidated(address indexed user, uint256 beforeCollateralAmount, uint256 afterCollateralAmount, uint256 beforeDebtAmount);
+
+    constructor(address _eETH, address _USDC, address _etherFiCashSafe) {
+        eETH = IERC20(_eETH);
+        USDC = IERC20(_USDC);
+        etherFiCashSafe = _etherFiCashSafe;
+    }
+
+    function supplyUSDC(uint256 amount) external {
+        SafeERC20.safeTransferFrom(USDC, msg.sender, address(this), amount);
+
+        emit SuppliedUSDC(amount);
+    }
+
+    function depositEETH(address user, uint256 amount) external {
+        _collaterals[user] += amount;
+        totalCollateralAmount += amount;
+
+        SafeERC20.safeTransferFrom(eETH, msg.sender, address(this), amount); // use safeTransferFrom
+
+        emit DepositedCollateral(user, amount);
+    }
+
+    /// Users borrow USDC for payment using the deposited collateral eETH
+    /// - the user's borriwng amount is incremented by the exact `borrowUsdcAmount`
+    /// - the total borrowing amount is incremented by the exact `borrowUsdcAmount`
+    /// - the USDC is transferred to the `etherFiCashSafe`
+    function borrowUSDC(uint256 borrowUsdcAmount) external {
+        _borrowings[msg.sender] += borrowUsdcAmount;
+        totalBorrowingAmount += borrowUsdcAmount;
+
+        require(debtRatioOf(msg.sender) <= liquidationThreshold, "NOT_ENOUGH_COLLATERAL");
+        require(USDC.balanceOf(address(this)) >= borrowUsdcAmount, "INSUFFICIENT_LIQUIDITY");
+
+        SafeERC20.safeTransfer(USDC, etherFiCashSafe, borrowUsdcAmount);
+
+        emit Borrowed(msg.sender, borrowUsdcAmount);
+    }
+
+    /// Users repay the borrowed USDC in USDC
+    function repayWithUSDC(uint256 repayUsdcAmount) external {
+        _borrowings[msg.sender] -= repayUsdcAmount;
+        totalBorrowingAmount -= repayUsdcAmount;
+
+        SafeERC20.safeTransferFrom(USDC, msg.sender, address(this), repayUsdcAmount);
+
+        emit RepaidWithUSDC(msg.sender, repayUsdcAmount);
+    }
+
+    /// Users repay the borrowed USDC in eETH
+    /// Equivalent to partially liquidating the debt using the collateral
+    function repayWithEETH(uint256 repayUsdcAmount) external {
+        _repayWithEEthCollateral(msg.sender, repayUsdcAmount);
+    }
+
+    // https://docs.aave.com/faq/liquidations
+    /// Liquidate the user's debt by repaying the entire debt using the collateral
+    /// @dev do we need to add penalty?
```

Comment:
I don't think so. The system should be designed in a way that our protocol's position never actually get liquidated incurring fees from aave.

Reply:
our protocol's position in L1 Aave shouldn't be liquidiated. that is our responsibility

this function is for the cash users debt on L2